### ISSUE TITLE
Bitbucket Credentials is stored as plain-text

### ISSUE THREAD

**Issue by firelite** [afragen/github-updater#695]
I'm really liking GitHub Updater so far but I noticed that Bitbucket Private settings stores my Bitbucket Username and Password as plain-text.  Is there any way to create secure connections to my private theme repos, without leaving my passwords on the websites I manage?  I think GitHub access/repo token is a great idea.

**Comment by afragen** [comment]
tl;dr - use an app password.

Unless I can figure out how to use oAuth, I have to use Basic Auth for Bitbucket Authentication. Bitbucket doesn’t provide a token method similar to others. The username/password are stored in the options table. Basic Auth requires a plaintext hashed user:password. I hate to tell you that the real security risk isn’t your database as much as someone being able to scrape the HTTP headers. I mitigate as much as possible by only loading on admin pages that are required and only for admin-level users.

They do allow for creating a [app password](https://confluence.atlassian.com/bitbucket/app-passwords-828781300.html).

It might be more trouble, but you could have your client create a Bitbucket account and grant their user `read` access to your specific repos.

If you want to help with a PR for oAuth it would be great and help solve this problem for everyone.
